tests/e2e/configuration/configuration_test.go

```diff
@@ -0,0 +1,287 @@
+}
+
+type Item struct {
+	Value    string            `json:"value,omitempty"`
+	Version  string            `json:"version,omitempty"`
+	Metadata map[string]string `json:"metadata,omitempty"`
+}
+
+type appResponse struct {
+	Message   string `json:"message,omitempty"`
+	StartTime int    `json:"start_time,omitempty"`
+	EndTime   int    `json:"end_time,omitempty"`
+}
+
+func TestMain(m *testing.M) {
+	utils.SetupLogs("configuration e2e")
+	utils.InitHTTPClient(true)
+
+	p := os.Getenv(componentNameEnvVar)
+	if p != "" {
+		componentName = p
+	}
+
+	// These apps will be deployed for hellodapr test before starting actual test
+	// and will be cleaned up after all tests are finished automatically
+	testApps := []kube.AppDescription{
+		{
+			AppName:           "configurationapp",
+			DaprEnabled:       true,
+			ImageName:         "e2e-configurationapp",
+			Replicas:          1,
+			IngressEnabled:    true,
+			MetricsEnabled:    true,
+			DaprMemoryLimit:   "200Mi",
+			DaprMemoryRequest: "100Mi",
+			AppMemoryLimit:    "200Mi",
+			AppMemoryRequest:  "100Mi",
+		},
+	}
+
+	log.Printf("Creating TestRunner\n")
+	tr = runner.NewTestRunner("configuration", testApps, nil, nil)
+	os.Exit(tr.Start(m))
+}
+
+var configurationTests = []struct {
+	name    string
+	handler func(t *testing.T, appExternalUrl string)
+}{
+	{
+		name:    "testGet",
+		handler: testGet,
+	},
+	{
+		name:    "testSubscribe",
+		handler: testSubscribe,
+	},
+	{
+		name:    "testUnsubscribe",
+		handler: testUnsubscribe,
+	},
+}
+
+// Generates key-value pairs
+func generateKeyValues(keyCount int, version string) map[string]*Item {
+	m := make(map[string]*Item, keyCount)
+	k := counter
+	for ; k < counter+keyCount; k++ {
+		key := runID + "-key-" + strconv.Itoa(k)
+		val := runID + "-val-" + strconv.Itoa(k)
+		m[key] = &Item{
+			Value:    val,
+			Version:  version,
+			Metadata: map[string]string{},
+		}
+	}
+	counter = k
+	return m
+}
+
+// Updates `mymap` with new values for every key
+func updateKeyValues(mymap map[string]*Item, version string) map[string]*Item {
+	m := make(map[string]*Item, len(mymap))
+	k := counter
+	for key := range mymap {
+		updatedVal := runID + "-val-" + strconv.Itoa(k)
+		m[key] = &Item{
+			Value:    updatedVal,
+			Version:  version,
+			Metadata: map[string]string{},
+		}
+		k++
+	}
+	counter = k
+	return m
+}
+
+// returns the keys of a map
+func getKeys(mymap map[string]*Item) []string {
+	keys := []string{}
+	for key := range mymap {
+		keys = append(keys, key)
+	}
+	return keys
+}
+
+func testGet(t *testing.T, appExternalUrl string) {
+	updateUrl := fmt.Sprintf("http://%s/update-key-values", appExternalUrl)
+	items := generateKeyValues(10, v1)
+	itemsInBytes, _ := json.Marshal(items)
+	resp, statusCode, err := utils.HTTPPostWithStatus(updateUrl, itemsInBytes)
+	require.NoError(t, err, "error updating key values")
+	require.Equal(t, 200, statusCode, "expected statuscode 200, got %d. Error: %s", statusCode, string(resp))
```

Comment:
When you put a `%` in the format and use `require/assert`, you must add `f` at the end of the function. For example: `require.Equalf`. Oterhwise it doesn't use sprintf internally and logs come out with some really weird formatting